tlv/fuzz_test.go

```diff
@@ -168,3 +168,111 @@ func FuzzTUint64(f *testing.F) {
 		harness(t, data, ETUint64, DTUint64, &val, uint64(decodeLen))
 	})
 }
+
+// encodeParsedTypes re-encodes TLVs decoded from a stream, using the
+// parsedTypes and decodedRecords produced during decoding.
+func encodeParsedTypes(t *testing.T, parsedTypes TypeMap,
+	decodedRecords []Record) []byte {
+
+	var encodeRecords []Record
+	for typ, val := range parsedTypes {
+		if typ < Type(len(decodedRecords)) {
```

Comment:
would be helpful if we add some docs explaining why this check.

Reply:
Done in https://github.com/lightningnetwork/lnd/pull/7889/commits/5bb4df424c2a2a926b881449c9585fed6174e6da.